### PR TITLE
Informing the user about the HEX conversion scheme of Integers

### LINKED ISSUE #15136: Hex converter giving invalid results
### Describe the bug
The hexadecimal converter 'HEX' in the insert field is giving invalid results. Suppose if I enter 70 as my int value it should give answer as '46' in HEX but it is giving answer as '3730'.
The 'OCT' converter for the same is working correct as I verified.
### To Reproduce
1.  Add int type column in any table.
2. Go to insert section.
3. Select the function 'HEX' in that 'int' column.
4. Enter some int value and observe the result in the added column
### Expected behaviour
The 'HEX' function should give out the corrected value.
### Screenshots
Screenshot 2019-03-31 at 10 47 55 PM
This value '3730' was received from type conversion of int '70' to hex
### Server configuration

Operating system: macOS Server(5.7.1 )
Web server: Apache/2.4.37 (Unix) OpenSSL/1.0.2q PHP/7.3.0 mod_perl/2.0.8-dev Perl/v5.16.3
Database version: Localhost via UNIX socket
PHP version: 7.3.0
phpMyAdmin version: 4.8.4

### Client configuration
Browser: Safari (12.0.3)
Operating system: macOS Mojave (version 10.14.2)

### PR DESCRIPTION
**Pull request by thesmallstar**
Signed-off-by: Manthan Surkar <manthan.surkar@gmail.com>

### Description

Please describe your pull request.
A note is shown to the user when he tries to apply HEX function on integer.
Cases, like adding multiple integer notes and removing them, is also taken care of. 

image

Fixes #15136

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: js/table/change.js [hunk 1]
```diff
@@ -196,7 +196,26 @@ function verificationsAfterFieldChange (urlField, multiEdit, theType) {
...
     }
-
+    //Add note when HEX function is selected on a int
+//Add note when HEX function is selected on a int
+    var newHexInfo =
+     "<br><p id=note" +
+        target.id +
+       ">" +
+      "HEX function will treat the integer as String while calculating hexadecimal value" +
+      "</p>";
```
Comment (williamdes) [inline js/table/change.js]: Can you add the string to `js/messages.php` and use it here using `Messages.`

**Review by williamdes — CHANGES_REQUESTED** [review]
I am not sure if adding html that way is the best way to do, but I also do not have a better solution today

File: js/table/change.js [hunk 2]
```diff
@@ -197,6 +197,25 @@ function verificationsAfterFieldChange (urlField, multiEdit, theType) {
         });
     }
 
+    //Add note when HEX function is selected on a int
+    var newHexInfo =
```
Comment (williamdes) [inline js/table/change.js]: can you make it oneline

File: js/messages.php [hunk 3]
```diff
@@ -541,6 +541,7 @@ function () {
 $js_messages['strYes'] = __('Yes');
 $js_messages['strCopyEncryptionKey'] = __('Do you want to copy encryption key?');
 $js_messages['strEncryptionKey'] = __('Encryption key');
+$js_messages['HexConversionInfo'] = __('HEX function will treat the integer as String while calculating hexadecimal value');
```
Comment (williamdes) [inline js/messages.php]: `String` should not have a capital letter ?
```suggestion
/* l10n: Please explain here to translators what they are translating 
$js_messages['HexConversionInfo'] = __('HEX function will treat the integer as string while calculating hexadecimal value');
```

**Comment by thesmallstar** [comment]
@williamdes, Since this is not an error we cannot put this string/note in any error code section.
The following way is referred from the same file, selecting AES_ENCRYPT adds HTML to provide similar functionality. 

I also couldn't find a better way :P, Hope this works.

**Review by williamdes — CHANGES_REQUESTED** [review]
> I also couldn't find a better way :P, Hope this works.
> The following way is referred from the same file

Okay then :p

You are on the right path ;)
Be sure to run `yarn run js-lint`/`yarn run js-lint --fix`

**Comment by thesmallstar** [comment]
@williamdes Thank you so much for the suggestions, helps us improve :)

File: js/messages.php [hunk 4]
```diff
@@ -541,6 +541,8 @@ function () {
 $js_messages['strYes'] = __('Yes');
 $js_messages['strCopyEncryptionKey'] = __('Do you want to copy encryption key?');
 $js_messages['strEncryptionKey'] = __('Encryption key');
+/*Tip for HEX conversion of Integers*/
```
Comment (williamdes) [inline js/messages.php]:
```suggestion
/* l10n: Tip for HEX conversion of Integers*/
```

**Review by williamdes — APPROVED** [review]

File: js/messages.php (hunk 4 above)
Comment (williamdes) [reply js/messages.php]: @thesmallstar If you remove the comment prefix `gettext` will not extract it

Comment (thesmallstar) [reply js/messages.php]: on it :D

**Comment by williamdes** [comment]
Looks nice, I would like a team review or point of view before merging

**Comment by kartik1000** [comment]
@thesmallstar You can add the notice in <strong> tags to highlight it. Also can't we add a notice only once when the function is selected in more than one columns. Don't makes much sense to keep repeating it again and again. What do you feel @williamdes? Rest everything looks good 😃

**Comment by williamdes** [comment]
What about adding a validator the adds the message if the input is not considered correct or risky ?

Like the error message when typing text in an int

**Comment by kartik1000** [comment]
Yess @williamdes that would not confuse a new user in case the output is correct and we are still giving some notice.  That would be the best.

**Comment by williamdes** [comment]
I think you will need to use master branch because it seems more like an enhancement to me

**Comment by thesmallstar** [comment]
@willamdes okay, I am still trying to find the best way to inform the user, I will let you know. @kartik1000 do you have more suggestions?

**Comment by kartik1000** [comment]
@thesmallstar I think if you can make the change @williamdes has mentioned that would be the best and it won't require any more changes

**Review by MauricioFauth — APPROVED** [review]
Thanks for your contribution @thesmallstar.

LGTM.
I think this notice should be added in the _master_ branch.

**Comment by williamdes** [comment]
Thank you @thesmallstar !

**Comment by williamdes** [comment]
You will need to force push QA_5_0 to the right commit using #15942

**Pull request merged**